core/pta/attestation.c

```diff
@@ -330,7 +330,8 @@ static TEE_Result init_key(void)
 		 * e is 65537 so its bignum size is 3 bytes. d and n can be up
 		 * to MAX_KEY_SIZE bits.
 		 */
-		size = bufsize(3, MAX_KEY_SIZE / 8, MAX_KEY_SIZE / 8);
+		size = bufsize(3, CFG_ATTESTATION_PTA_KEY_SIZE / 8,
+			       CFG_ATTESTATION_PTA_KEY_SIZE / 8);
```

Comment:
Wouldn't it be better to change the assertion instead? I mean `CFG_ATTESTATION_PTA_KEY_SIZE` should be understood as the size of the key in case it needs to be generated only, otherwise whatever has been saved previously has the priority. That would be more friendly to existing devices in case the TEE is upgraded and `CFG_ATTESTATION_PTA_KEY_SIZE` has been changed in the mean time.

Reply:
You're right, I've changed the condition to ensure retro-compatibility.
Are you okay with always allocating the maximum size for the key?

Reply:
> Are you okay with always allocating the maximum size for the key?

Seems acceptable, yes.